pkg/dev/dev.go

```diff
@@ -0,0 +1,128 @@
+// This file is part of MinIO Direct CSI
+// Copyright (c) 2020 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package dev
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+)
+
+type BlockDevice struct {
```

Comment:
We could also need to store the devpath in order to query more information later. The devpath is the path to the device directory relative to `/sys`, without any symlink elements and with leading `/`. It is used by the kernel as a unique key to the device. Example devpath:

```
/devices/pci0000:00/0000:00:1d.4/0000:3d:00.0/nvme/nvme0/nvme0n1
```

Reply:
There is no value in it. This is the physical port where the drive is attached. If the admin changes the slots, then this path will change but it won’t matter if we use the file system uuid.

Reply:
Of course the value is not constant for the physical drive, but that was not my intention. My intention is that you might need the devpath in order to query the kernel API for the attributes I mentioned in my other comment.